Source/DataSources/RectangleGeometryUpdater.js

```diff
@@ -113,17 +134,10 @@ define([
             if (!defined(currentColor)) {
                 currentColor = Color.WHITE;
             }
-            color = ColorGeometryInstanceAttribute.fromColor(currentColor);
-            attributes = {
-                show : show,
-                distanceDisplayCondition : distanceDisplayConditionAttribute,
-                color : color
-            };
-        } else {
-            attributes = {
-                show : show,
-                distanceDisplayCondition : distanceDisplayConditionAttribute
-            };
+            attributes.color = ColorGeometryInstanceAttribute.fromColor(currentColor);
+        }
+        if (defined(this._options.offsetAttribute)) {
```

Comment:
Why not move this into the attributes initialization?  Would keep the shape of the object the shame, also initialize color to undefined as well (for the same reason),  This way attributes always has the same properties.

Reply:
Why does this matter?  I'm pretty sure the result would be the same regardless.

Reply:
It's so browsers can better optimize the code.  If `attributes` always has the same set of properties (even if some are undefined) the browser can do a better job of optimizing compared to `attributes` sometimes having an offset property and sometimes not.